### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by erdoganishe**
Refactor circom pairing lib

### CODE REVIEW

File: circuits/ecdsa/brainpoolP256r1/circomPairing/blsSignature.circom [hunk 1]
```diff
@@ -22,40 +22,58 @@ template CoreVerifyPubkeyG1NoCheck(CHUNK_SIZE, CHUNK_NUMBER){
...
+    var total = 12 * CHUNK_NUMBER;
+    for(var i = 0; i < 6; i++){
+        for(var j = 0; j < 2; j++){
+            for(var idx = 0; idx < CHUNK_NUMBER; idx++){
+                is_valid[i][j][idx] = IsZero(); 
+                if(i= = 0 && j= = 0 && idx= = 0){
+                    is_valid[i][j][idx].in <== finalExp.out[i][j][idx] - 1;
+                } else {
+                    is_valid[i][j][idx].in <== finalExp.out[i][j][idx];
+                    total -= is_valid[i][j][idx].out;
```
Comment (1KitCat1) [inline circuits/ecdsa/brainpoolP256r1/circomPairing/blsSignature.circom:71]: Error during refactor, invalid logic @erdoganishe

File: circuits/ecdsa/brainpoolP256r1/circomPairing/blsSignature.circom [hunk 2]
```diff
@@ -22,40 +22,58 @@ template CoreVerifyPubkeyG1NoCheck(CHUNK_SIZE, CHUNK_NUMBER){
...
-            is_valid[i][j][idx].in <== finalExp.out[i][j][idx] - 1;
-        else
-            is_valid[i][j][idx].in <== finalExp.out[i][j][idx];
-        total -= is_valid[i][j][idx].out; 
+    var total = 12 * CHUNK_NUMBER;
+    for(var i = 0; i < 6; i++){
+        for(var j = 0; j < 2; j++){
+            for(var idx = 0; idx < CHUNK_NUMBER; idx++){
+                is_valid[i][j][idx] = IsZero(); 
+                if(i= = 0 && j= = 0 && idx= = 0){
```
Comment (1KitCat1) [inline circuits/ecdsa/brainpoolP256r1/circomPairing/blsSignature.circom:67]: i= = 0
?
@erdoganishe

Comment (erdoganishe) [reply circuits/ecdsa/brainpoolP256r1/circomPairing/blsSignature.circom:67]: done

File: circuits/ecdsa/brainpoolP256r1/circomPairing/blsSignature.circom (hunk 1 above)
Comment (erdoganishe) [reply circuits/ecdsa/brainpoolP256r1/circomPairing/blsSignature.circom:71]: Done

**Pull request merged**